### PR TITLE
don-862 implement segmented control

### PR DESCRIPTION
**Pull request by LokmaneKrizou**
+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-android/blob/main/CONTRIBUTING.md)

Remember to include the following changes:
+ [x] Component `README.md`
+ [x] Tests

If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)

### CODE REVIEW

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/segmentedcontrol/BpkSegmentedControl.kt [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    SurfaceContrast,
+}
+
+private const val maxNumberOfButtons = 4
+
+@Composable
+fun BpkSegmentedControl(
+    buttonContents: List<String>,
+    onItemClick: (Int) -> Unit,
+    selectedInt: Int,
```
Comment (marianeum) [inline backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/segmentedcontrol/BpkSegmentedControl.kt:38]: `selectedIndex`

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/segmentedcontrol/internal/BpkSegmentedControlColors.kt [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.compose.segmentedcontrol.internal
+
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.graphics.Color
+import net.skyscanner.backpack.compose.utils.dynamicColorOf
+
+internal val surfaceContrastSegmentedControlOn: Color
```
Comment (marianeum) [inline backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/segmentedcontrol/internal/BpkSegmentedControlColors.kt:25]: where do these come from? there's already a `BpkSegmentedControlColors` file which is the custom colours which i think should be used here?

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/segmentedcontrol/internal/BpkSegmentedControlImpl.kt [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+import net.skyscanner.backpack.compose.segmentedcontrol.BpkSegmentedControlStyle
+import net.skyscanner.backpack.compose.text.BpkText
+import net.skyscanner.backpack.compose.theme.BpkTheme
+import net.skyscanner.backpack.compose.tokens.BpkElevation
+import net.skyscanner.backpack.compose.tokens.BpkSpacing
+
+private const val BpkAnimationDurationMs = 50
+private const val RowNumber = 1
+private val BpkSegmentedControlHeight = 32.dp
+private val ButtonShape = RoundedCornerShape(8.dp)
```
Comment (marianeum) [inline backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/segmentedcontrol/internal/BpkSegmentedControlImpl.kt:59]: `BpkBorderRadius.Sm`

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/segmentedcontrol/internal/BpkSegmentedControlImpl.kt [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+    selectedInt: Int,
+    shadow: Boolean,
+    type: BpkSegmentedControlStyle,
+    modifier: Modifier = Modifier,
+) {
+    Row(
+        verticalAlignment = Alignment.CenterVertically,
+        modifier = modifier
+            .height(BpkSegmentedControlHeight)
+            .semantics { collectionInfo = CollectionInfo(rowCount = RowNumber, columnCount = buttonContents.size) }
```
Comment (marianeum) [inline backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/segmentedcontrol/internal/BpkSegmentedControlImpl.kt:74]: you might also need `selectableGroup` here (or instead)

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/segmentedcontrol/internal/BpkSegmentedControlImpl.kt [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+            textAlign = TextAlign.Center,
+            color = getTextColor(type, isSelected),
+            style = BpkTheme.typography.footnote,
+            overflow = TextOverflow.Ellipsis,
+            maxLines = 1,
+        )
+        if (showDivider) {
+            BpkDivider(
+                modifier = Modifier
+                    .width(1.dp)
```
Comment (marianeum) [inline backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/segmentedcontrol/internal/BpkSegmentedControlImpl.kt:147]: `BpkBorderSize.Sm`

**Review by marianeum — COMMENTED** [review]
just a few minor comments

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/segmentedcontrol/internal/BpkSegmentedControlColors.kt (hunk 2 above)
Comment (LokmaneKrizou) [reply backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/segmentedcontrol/internal/BpkSegmentedControlColors.kt:25]: noice, didn't notice that till I rebased 😓

Comment (LokmaneKrizou) [reply backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/segmentedcontrol/internal/BpkSegmentedControlColors.kt:25]: actually they were there even before 😅

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/segmentedcontrol/internal/BpkSegmentedControlImpl.kt [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+            .applyBackground(
+                selected = isSelected,
+                background = getButtonColor(type),
+            )
+            .padding(horizontal = BpkSpacing.Md)
+            .selectable(
+                selected = isSelected,
+                role = Role.RadioButton,
+                interactionSource = remember { MutableInteractionSource() },
+                indication = null,
```
Comment (marianeum) [inline backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/segmentedcontrol/internal/BpkSegmentedControlImpl.kt:132]: is it intentional that there's no ripple? we should always have some touch feedback

**Review by marianeum — COMMENTED** [review]
one small comment, otherwise looks good

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/segmentedcontrol/internal/BpkSegmentedControlImpl.kt (hunk 6 above)
Comment (LokmaneKrizou) [reply backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/segmentedcontrol/internal/BpkSegmentedControlImpl.kt:132]: at first I added it, then I removed it while playing with shadow, good spot

**Review by marianeum — APPROVED** [review]

**Pull request merged**